app/javascript/components/widget-display.test.jsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import ReactTestRenderer from 'react-test-renderer';
+import fetch from 'unfetch';
+import { ApolloClient } from 'apollo-client';
+import { ApolloLink } from 'apollo-link';
+import { HttpLink } from 'apollo-link-http';
+import {
+  InMemoryCache,
+  IntrospectionFragmentMatcher,
+} from 'apollo-cache-inmemory';
+import ActionCable from 'actioncable';
+import { getMainDefinition } from 'apollo-utilities';
+import ActionCableLink from 'graphql-ruby-client/subscriptions/ActionCableLink';
+import { persistCache } from 'apollo-cache-persist';
+import helioSchema from '@javascript/schema.json';
+import { WidgetDisplay } from '@components/widget-display';
+import { ApolloProvider } from 'react-apollo';
+
+const cable = ActionCable.createConsumer('/cable');
+
+const httpLink = new HttpLink({
```

Comment:
Sorry I think I sent you down the wrong path here. A snapshot test should just render a component without connecting to any external resources. Lets pair to work through this.